### PR TITLE
Implement meaningful linting and syntax checking

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This updates our implementation of both CodeClimate and GitHub Actions to have better coordination around their use of the PHP CodeSniffer (PHPCS) tool for static code analysis.

This updates `phpcs.xml` to apply the WordPress coding standards, in their entirety, even to locally-developed plugins (achieved by changing from a global exemption for all plugin code to a defined list of plugins - those added by Composer). Problems flagged by this tool can be discussed during code review, and should the proposer and reviewer agree to accept some non-conforming code, CodeClimate would then not warn about these issues until those lines of code are changed again.

GitHub Actions, meanwhile, uses different ruleset - `phpcs.security.xml` - inherited from `phpcs.xml` but with all non-security sniffs disabled. GitHub Actions also evaluates all PHP files for syntax problems, which should prevent the easiest-to-avoid problems like forgotten semicolons or other fundamentally broken code. This will be done for the entire codebase on every PR. Syntax problems will always be highlighted, while violations of the WordPress security standard can only be whitelisted by explicit code comments (which should obviously be used sparingly, but may be unavoidable with some legacy code).

As a result of these changes, this PR also updates the themes already added to the repo to comply with these standards.

We have already added two local plugins to this repo, but they are still excluded from scanning for now. The next PR will start scanning these, and fix their issues.

**To see these updates in action when non-compliant code is encountered, please see the demo PR at #21 **

### Secrets

- [ ] All new secrets have been added to Pantheon tiers
- [ ] Relevant secrets have been updated in Github Actions
- [ ] All new secrets documented in README
- [x] No secrets need to be changed

### Documentation

- [x] Project documentation has been updated
- [ ] No documentation changes are needed

### Accessibility

- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)

### Stakeholder approval

- [ ] Stakeholder approval has been confirmed
- [x] Stakeholder approval is not needed

### Dependencies

YES dependencies are updated

## Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] The changes have been verified
- [x] The documentation has been updated or is unnecessary
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: config/application.php [hunk 1]
```diff
@@ -116,21 +116,6 @@
 // Limit the number of post revisions that Wordpress stores (true (default WP): store every revision)
 Config::define('WP_POST_REVISIONS', env('WP_POST_REVISIONS') ?: true);
 
-/**
```
Comment (JPrevost) [inline config/application.php:119]: How are these changes related to the linting configuration?

File: phpcs.security.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Security Standards">
```
Comment (JPrevost) [inline phpcs.security.xml:2]: I wonder if it would be helpful to name this in a way to remind us that this is both Security and Syntax (if I'm reading the notes correctly on what is going on here)?

File: config/application.php (hunk 1 above)
Comment (matt-bernhardt) [reply config/application.php:119]: The change in this file was prompted by CodeClimate when I pushed the configuration as part of this work. This was the code block that got moved by both us and Pantheon, and it looks like when we resolved their changes we ended up with two of them. CodeClimate caught the duplication.

We didn't specifically talk last week about whether to use the duplication check on CodeClimate or not, but it has been part of how we use the platform in the past, and given how CodeClimate behaves, it seems like a good tool to use selectively (rather than on GitHub, which is global).

File: phpcs.security.xml (hunk 2 above)
Comment (matt-bernhardt) [reply phpcs.security.xml:2]: This file is just the security side of that check. The syntax part of that check is performed by the other line in the `composer test` script - https://github.com/MITLibraries/mitlib-wp-network/pull/20/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R154

File: config/application.php (hunk 1 above)
Comment (JPrevost) [reply config/application.php:119]: Got it thanks.

For future consideration, noting this removal as a side effect of implementing code climate would have helped me as a reviewer understand wtf was going on. (No change is requested now as we've now talked about it so I understand the removal).

File: phpcs.security.xml (hunk 2 above)
Comment (JPrevost) [reply phpcs.security.xml:2]: Ah got it thanks. So _that's_ what that gibberish line of code does lol.

**Review by JPrevost — APPROVED** [review]
